jerry-core/jrt/jrt-bit-fields.h

```diff
@@ -16,7 +17,29 @@
 #ifndef JERRY_BIT_FIELDS_H
 #define JERRY_BIT_FIELDS_H
 
-extern uint64_t __attr_const___ jrt_extract_bit_field (uint64_t, size_t, size_t);
-extern uint64_t __attr_const___ jrt_set_bit_field_value (uint64_t, uint64_t, size_t, size_t);
+/**
+ * Extract a bit-field.
+ *
+ * @param type type of container
+ * @param container container to extract bit-field from
+ * @param lsb least significant bit of the value to be extracted
+ * @param width width of the bit-field to be extracted
+ * @return bit-field's value
+ */
+#define JRT_EXTRACT_BIT_FIELD(type, container, lsb, width) \
+  (((container) >> lsb) & ((((type) 1) << (width)) - 1))
```

Comment:
I had two reasons to remove the asserts:
* I didn't know how to insert them in the macro (which is an expression only, not a block of statements), and
* during the first experiments with the patch the compiler *did* emit warnings (which were turned to errors) when `lsb` and/or `width` was 'incompatible' with the type of `container`, and I've considered them as good replacements of the asserts.

As for the `1u`s: in the first case, it is certainly superfluous, since `1` is cast to `type`. In the second case, it might also be better to type-cast it instead of using the hard-coded `1u`. IMHO.